cypress/e2e/generate_prompt.cy.js

```diff
@@ -1,27 +1,59 @@
-describe('Generate a prompt and check output', () => {
+describe('Prompt generation', () => {
     const output_locator = 'textarea[placeholder*="AI generated content"]';
     const category_input = 'category input from cypress';
     const prompt_input = 'prompt input from cypress';
-
-    it('Opens the page', () => {
+    beforeEach(() => {
         cy.visit('http://localhost:3000/');
     });
-    it('Write text in category prompt', () => {
+
+    it('should generate a prompt and get result', () => {
         cy.get('input[placeholder*="category"]').type(category_input);
-    });
-    it('Write user input for first prompt', () => {
         cy.get('textarea[placeholder*="User input here"]').type(prompt_input);
-    });
-    it('Output box should be empty', () => {
         cy.get(output_locator).invoke('text').should('be.empty');
-    });
-    it('Push Generate button', () => {
         cy.get('button:contains("Generate")').first().click();
-    });
-    it('Output box should have text', () => {
         cy.get(output_locator).should(
             'include.text',
             `Theme: ${category_input}\\n${prompt_input}`
         );
     });
+
+    it('should have generate button locked without prompt input', () => {
+        cy.get(output_locator).invoke('text').should('be.empty');
+        cy.get('button:contains("Generate")').first().should('be.disabled');
+    });
+
+    it('should generate all boxes with generate all button', () => {
```

Comment:
Code looks fine, maybe the readability could be increased by small comments in the code (?)
e.g.,
```javascript
       // At start only 1 prompt where output empty
        cy.get('input[placeholder*="category"]').type(category_input);
        cy.get(output_locator).invoke('text').should('be.empty');
        cy.get('[data-testid="prompt"]').should('have.length', 1);
        // Adding new prompts
        for (let i = 1; i <= numExtra; i++) {
            cy.get('[data-testid="fab-button"]').click();
        }
        // Added prompts should be visible
        cy.get('[data-testid="prompt"]').should('have.length', 1 + numExtra);
```

Reply:
Resolved by https://github.com/bytecraftoy/aalto-2022/pull/53/commits/a8014f04b42f36d0872b4252104e0deafd4f14c4